packages/core/src/external/aws/lambda-logic/document-uploader.ts

```diff
@@ -75,15 +94,59 @@ export async function documentUploaderHandler(
   }
 }
 
-async function forwardCallToServer(cxId: string, apiServerURL: string, fileData: FileData) {
+async function forwardCallToServer(
+  cxId: string,
+  apiServerURL: string,
+  fileData: FileData
+): Promise<DocumentReference | undefined> {
   const url = `${apiServerURL}?cxId=${cxId}`;
   const encodedUrl = encodeURI(url);
 
   const resp = await api.post(encodedUrl, fileData);
   console.log(`Server response - status: ${resp.status}`);
   console.log(`Server response - body: ${resp.data}`);
+  return resp.data;
 }
 
 function buildDestinationKey(cxId: string, patientId: string, docId: string): string {
   return `${cxId}/${patientId}/${UPLOADS_FOLDER}/${cxId}_${patientId}_${docId}`;
 }
+
+async function createAndUploadMetadataFile(
+  s3Utils: S3Utils,
+  cxId: string,
+  patientId: string,
+  docId: string,
+  hash: string,
+  size: string,
+  docRef: DocumentReference
+): Promise<void> {
+  const createdTime = new Date().toISOString();
+  const uniquePatientId = createPatientUniqueId(cxId, patientId);
+  const title = docRef.description;
+  const classCode = docRef.type;
+  const practiceSettingCode = docRef.context?.practiceSetting;
+  const healthcareFacilityTypeCode = docRef.context?.facilityType;
+  const extrinsicObjectXml = createExtrinsicObjectXml({
+    createdTime,
+    hash,
+    repositoryUniqueId: METRIPORT_REPOSITORY_UNIQUE_ID,
+    homeCommunityId: METRIPORT_HOME_COMMUNITY_ID,
+    size,
+    patientId: uniquePatientId,
+    classCode,
+    practiceSettingCode,
+    healthcareFacilityTypeCode,
+    documentUniqueId: docId,
+    title,
+  });
+
+  const s3KMetadataFileName = buildDestinationKeyMetadata(cxId, patientId, docId);
```

Comment:
nit: confusing var name - what's `K`? Took me a second to realize it probably stands for `Key`. Better to write it out to avoid breaking the reader's flow

Reply:
I think it was a typo 😆 fixing